### PR TITLE
[expo-cli] always use source maps to show better error stack traces

### PR DESCRIPTION
**Pull request by dsokal**
We include source maps in the npm package, but we don't use them unless a user sets `NODE_ENV` variable to `development`. I don't see any reason not to make use of source maps without setting that variable. I've also upgraded source maps related libraries and verified they work properly.

### CODE REVIEW

**Comment by fson** [comment]
Do you know what the runtime cost of this is? Does it only read the sourcemaps when an exception is thrown?

Since [we only show stack traces when `EXPO_DEBUG` is set](https://github.com/expo/expo-cli/blob/7dfb6290215c45eff0cb7a1e9649208df14615a5/packages/expo-cli/src/exp.js#L87-L91), I guess it would make sense to also only enable this when that environment variable is set. Or we might want to re-evaluate if hiding the stack traces like that even makes sense or if it would be better to always show them or at least write them to a `expo-debug.log` file in the current folder, perhaps.

**Comment by dsokal** [comment]
@fson 

I just checked and all source maps for `expo-cli` weight `396K`. It's not too much, so I think the point about runtime cost is rather irrelevant. 

Additionally, I quickly scrolled `source-map-support` and it looks like it's only loading source maps when needed (https://github.com/evanw/node-source-map-support/blob/master/source-map-support.js#L182). Also, `require('source-map-support').install();` doesn't do much more than setting (`Error.prepareStackTrace` function) - https://github.com/evanw/node-source-map-support/blob/master/source-map-support.js#L533

If we only show error (and stack traces) when `EXPO_DEBUG` is set then I agree we should only use `source-map-support` when it's set. However, I wouldn't be so sure about that. I don't have any particular example to show you, but I think I've seen errors' stack traces in `expo-cli` even when `EXPO_DEBUG` hadn't been set.

**Comment by fson** [comment]
Re: runtime cost, that sounds reasonable to me.

I guess it's possible for an error to slip through the `try-catch` I linked to in the commit above. What it tries to do though is to only show stack traces when this environment variable is set. (Which can be annoying at times.)

For now I'd only enable this when `EXPO_DEBUG` is set, and then a bit later work to make this better, so that we always write the errors to `expo-debug.log` (same ways e.g. npm or Yarn works) instead of logging to `~/.expo/log` and then get rid of the `EXPO_DEBUG` altogether.

**Comment by dsokal** [comment]
Hmmm, I made a change so that source maps support is being installed only when `EXPO_DEBUG` is set. Generally, I don't like this approach because as we both agreed there might be situations when an error is not caught by that `catch` when it occurs in a different place than `asyncAction` function. I wanted to always install source maps support because users sometimes send us error logs with stack traces and it's hard to read them when they have line numbers from transpiled files.

What is the reason you think we cannot / shouldn't always load source maps? We already agreed runtime cost is not a reason.

File: packages/expo-cli/bin/expo.js [hunk 1]
```diff
@@ -7,7 +7,9 @@ var ver = process.versions.node;
 ver = ver.split('-')[0]; // explode and truncate tag from version
 
 if (semver.satisfies(ver, '>=6.0.0')) {
-  require('source-map-support').install();
+  if (process.env.EXPO_DEBUG === 'true') {
```
Comment (ide) [inline packages/expo-cli/bin/expo.js]: Sometimes people will set EXPO_DEBUG=1, need to use something like getenv here: https://github.com/expo/expo-cli/blob/04f9572e091ce96c4b490afe94b8c5150de8fd25/packages/xdl/src/project/Doctor.js#L444

**Comment by dsokal** [comment]
@fson @ide do you guys approve this PR or you still think it's not good to merge?

**Pull request merged**